### PR TITLE
FIX(SLA): Finished SLA workflow

### PR DESCRIPTION
**Pull request by rosogon**
This PR finishes the DAM workflow and modifies the API for generating the template. Check README.md for examples on how to call the API.

### CODE REVIEW

**Comment by rosogon** [comment]
This PR also handles #164

**Comment by rosogon** [comment]
Please @MicheleGuerriero , @szenzaro : Could you have a look at it? If it looks good, please contact @andreaturli or @perezp to merge.

File: sla/sla-core/sla-service/src/main/java/eu/atos/sla/service/rest/SeacloudsRest.java [hunk 1]
```diff
@@ -199,16 +249,49 @@ public String createTemplate(SlaInfo slaInfo, boolean persist) throws JAXBExcept
     
     @POST
     @Path("agreements")
+    public Response createAgreement(
+            @Context UriInfo uriInfo, 
+            @QueryParam("agreementId") String agreementId,
+            @RequestBody String slaPayload)
```
Comment (MicheleGuerriero) [inline sla/sla-core/sla-service/src/main/java/eu/atos/sla/service/rest/SeacloudsRest.java]: is this "slaPayload" represented also in the sequence diagram?

File: sla/sla-core/sla-generator/src/test/java/eu/seaclouds/platform/sla/generator/AgreementGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        File file = new File(AgreementGeneratorTest.class.getResource("/template.xml").getFile());
+        String templateString = Files.toString(file, Charsets.UTF_8);
+        Template wsagTemplate;
+        try {
+            wsagTemplate = templateParser.getWsagObject(templateString);
+        } catch (ParserException e) {
+            throw new SlaGeneratorException(e.getMessage(), e);
+        }
+        Agreement wsagAgreement = new AgreementGenerator(wsagTemplate).generate();
+        JaxbUtils.print(wsagAgreement);
```
Comment (MicheleGuerriero) [inline sla/sla-core/sla-generator/src/test/java/eu/seaclouds/platform/sla/generator/AgreementGeneratorTest.java]: Is this test also asserting something?

File: sla/sla-core/sla-generator/src/test/java/eu/seaclouds/platform/sla/generator/SlaInfoBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.io.File;
+import java.io.IOException;
+
+import org.testng.annotations.Test;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+
+import eu.seaclouds.platform.sla.generator.SlaInfo.SlaInfoBuilder;
+
```
Comment (MicheleGuerriero) [inline sla/sla-core/sla-generator/src/test/java/eu/seaclouds/platform/sla/generator/SlaInfoBuilderTest.java]: Same as for the AgreementGeneratorTest

**Comment by MicheleGuerriero** [comment]
@rosogon I had a look and wrote some minor comments. Overall to me the PR looks fine. Please make sure that the sequence diagram is fully compliant with the service definition (for example it should be POST /seaclouds/commands/agreements if I'm not wrong), since I think this diagram could be really useful for who is going to use the SLA-service! Thanks.

**Comment by rosogon** [comment]
Thanks @MicheleGuerriero . I will make sure the sequence diagrams and the code are aligned.

**Comment by adriannieto** [comment]
As this involves the API Proxy in #202  what's the status of this?

**Comment by rosogon** [comment]
@adriannieto : The code in this PR is independent from #202 except in DateTimeAdapter, where you will have a conflict very easy to fix (the right lines are in 202).

I expect we can merge this.

**Comment by adriannieto** [comment]
If you want we could finish the review of the #202 and then merge this.

**Comment by rosogon** [comment]
Hi @MicheleGuerriero : I reviewed the sequence diagrams and they match the service except in one minor thing, already fixed.

**Pull request merged**